deploy/charts/firefly/README.md

```diff
@@ -0,0 +1,46 @@
+# FireFly
+
+A bare-bones Helm chart for installing a [FireFly](https://github.com/hyperledger-labs/firefly) node with robust templating of its configuration
+for development and production scenarios. Additionally, includes FireFly's [default DataExchange](https://github.com/hyperledger-labs/firefly-dataexchange-https) component
+for simple, private messaging using HTTPS backed with mTLS.
+
+## TL;DR
+
+```shell
+# Deploy a FireFly node w/ some dummy values
+$ helm install acme-firefly ./deploy/charts/firefly \
+  --set dataexchange.tlsSecret.name=acme-dx-tls \
+  --set config.organizationName=acme \
+  --set config.organizationIdentity="0xeb7284ce905e0665b7d42cabe31c76c45da1d331" \
+  --set config.fireflyContractAddress="0xeb7284ce905e0665b7d42cabe31c76c45da1d254"
+```
+
+> **Note**: FireFly requires additional configuration for its blockchain, database, and public storage in order to be fully functional. See [example-values.yaml](ci/it-values.yaml) for an example, and below for more details.
+
+## Introduction
+
+
+
+## Prerequisites
+
+* Kubernetes 1.14+
+* Helm 3.6.0
+* PV provisioner support in the underlying infrastructure
```

Comment:
Worth describing the dependency for Cert Manager when using DX a bit more?

I could see from the chart there's a self-signed cert generated with [genselfsignedcert](https://helm.sh/docs/chart_template_guide/function_list/#genselfsignedcert), but it seemed like there was some orchestration via `cert-manager` required for it to function. I wasn't 100% sure why.

Reply:
TL;DR I removed the `genSelfSignedCert` `Secret` as that would have only created confusion for when the user tried to autocreate a certificate via `certificate.enabled` but didn't have cert-manager installed. I updated the README to then say cert-manager is recommended but not required.

Long and short of it:

I was trying to use `not (.Capabilities.APIVersions.Has "cert-manager.io/v1")` to detect if a user had opted-in to using cert-manager but didn't actually have it installed and then I'd use `fail` to have the chart throw an error message. This worked great.

However, when linting the chart `not (.Capabilities.APIVersions.Has "cert-manager.io/v1")` did not work as expected due to https://github.com/helm/helm/issues/3377#issuecomment-704574852. It also required a K8s cluster be present to lint a chart which felt wrong. So linting with my `ci/it-values.yaml` was failing as a result.

To get around the linting failuring, originally I had the mess you saw with `genSelfSignedCert` as a fallback for when they set `certificate.enabled` but didn't have cert-manager installed. But I decided thats sort of a silent failure as that type of cert would not work (DX requires a `subject` in the cert which you can't provide via that method) so instead to make the linter happy w/o confusing the user, I have the `not (.Capabilities.APIVersions.Has "cert-manager.io/v1")` check adding a YAML comment to the manifest saying this will fail and I'm letting Helm/K8s itself handle the failure if the API doesn't exist i.e. cert-manager isn't installed. 🙃

Reply:
Thanks for the update, and the detailed background - very helpful (including as reference for others).